api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -185,6 +185,10 @@ message ExtAuthz {
   // When this field is true, Envoy will include the SNI name used for TLSClientHello, if available, in the
   // :ref:`tls_session<envoy_v3_api_field_service.auth.v3.AttributeContext.tls_session>`.
   bool include_tls_session = 18;
+
+  // Whether to increment cluster statistics (e.g. cluster.<cluster_name>.upstream_rq_*) on authorization failure.
+  // Prefixed with `no_` because proto bool defaults are `false`.
+  bool no_charge_cluster_response_stats = 19;
```

Comment:
why do not we default to `true` in code and call it `charge_cluster_response_stats` ? It is easier to read and follow

Reply:
proto3 optionals are not enabled in Envoy yet, proto3 custom defaults are not allowed, and proto3 default for bool is `false`. In particular, without `optional`, there's no way to detect from "set to false" and "default of false". In general I agree with you.

Reply:
It could be done with a BoolValue. @markdroth can you comment on the best way to specify this?

Reply:
I thought about doing it that way (with a BoolValue), but I prefer to avoid the pattern of "a comment in the proto API tells you what the code sets the default to". Happy to be overridden, though.

Reply:
The [API style guide](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md) suggests that `BoolValue` is the preferred choice:

> Use [wrapped scalar types](https://github.com/google/protobuf/blob/master/src/google/protobuf/wrappers.proto) if there is any potential need for a field to have a default value that does not match the proto3 defaults (0/false/""). For example, new features whose default value may change in the future or security mitigations that should be default safe in the future but are temporarily not enabled.

Reply:
Switched the API to BoolValue.